### ISSUE TITLE
RunPodSandbox fails on openSUSE (and crashes ocid)

### ISSUE THREAD

**Issue by cyphar** [kubernetes-incubator/cri-o#446]
This is a bit of a concern. I found this while working on getting cri-o to work on openSUSE.

```
DEBU[2017-04-12 08:04:51.602473588+10:00] RunPodSandboxRequest &RunPodSandboxRequest{Config:&PodSandboxConfig{Metadata:&PodSandboxMetadata{Name:sandbox2,Uid:,Namespace:,Attempt:1,},Hostname:,LogDirectory:,DnsConfig:nil,PortMappings:[],Labels:map[string]string{batch: no,type: small,},Annotations:map[string]string{daemon: ocid,owner: dragon,},Linux:&LinuxPodSandboxConfig{CgroupParent:,SecurityContext:nil,},},}
DEBU[2017-04-12 08:04:51.603054915+10:00] parsed reference into "[btrfs@/var/lib/ocid]docker.io/kubernetes/pause:latest"
DEBU[2017-04-12 08:04:52.228518392+10:00] created pod sandbox "4a50b57ceb4a4440b767f3a56697c675f98a0b9d5bbae55119880f8863547b22"
DEBU[2017-04-12 08:04:52.278850598+10:00] pod sandbox "4a50b57ceb4a4440b767f3a56697c675f98a0b9d5bbae55119880f8863547b22" has work directory "/var/lib/ocid/btrfs-containers/4a50b57ceb4a4440b767f3a56697c675f98a0b9d5bbae55119880f8863547b22/userdata"
DEBU[2017-04-12 08:04:52.278930453+10:00] pod sandbox "4a50b57ceb4a4440b767f3a56697c675f98a0b9d5bbae55119880f8863547b22" has run directory "/var/run/containers/storage/btrfs-containers/4a50b57ceb4a4440b767f3a56697c675f98a0b9d5bbae55119880f8863547b22/userdata"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x1 pc=0x494e73]

goroutine 15 [running]:
panic(0xbc1ae0, 0xc4200100d0)
        /usr/lib64/go/src/runtime/panic.go:500 +0x1a1
github.com/kubernetes-incubator/cri-o/server.(*Server).RunPodSandbox(0xc42036e780, 0x7ff2b0c42238, 0xc4202e4de0, 0xc4204de260, 0x0, 0x0, 0x0)
        /home/cyphar/.local/src/github.com/kubernetes-incubator/cri-o/server/sandbox_run.go:196 +0xb63
github.com/kubernetes-incubator/cri-o/vendor/k8s.io/kubernetes/pkg/kubelet/api/v1alpha1/runtime._RuntimeService_RunPodSandbox_Handler(0xc93060, 0xc42036e780, 0x7ff2b0c42238, 0xc4202e4de0, 0xc4203599f0, 0x0, 0x0, 0x0, 0x43ae48, 0xc42001400c)
        /home/cyphar/.local/src/github.com/kubernetes-incubator/cri-o/vendor/k8s.io/kubernetes/pkg/kubelet/api/v1alpha1/runtime/api.pb.go:2300 +0x27d
github.com/kubernetes-incubator/cri-o/vendor/google.golang.org/grpc.(*Server).processUnaryRPC(0xc420404a00, 0x12bfdc0, 0xc4200ae6c0, 0xc4204672c0, 0xc42022c5a0, 0x12a7638, 0xc4202e4db0, 0x0, 0x0)
        /home/cyphar/.local/src/github.com/kubernetes-incubator/cri-o/vendor/google.golang.org/grpc/server.go:608 +0xc50
github.com/kubernetes-incubator/cri-o/vendor/google.golang.org/grpc.(*Server).handleStream(0xc420404a00, 0x12bfdc0, 0xc4200ae6c0, 0xc4204672c0, 0xc4202e4db0)
        /home/cyphar/.local/src/github.com/kubernetes-incubator/cri-o/vendor/google.golang.org/grpc/server.go:766 +0x6b0
github.com/kubernetes-incubator/cri-o/vendor/google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc4202ca5a0, 0xc420404a00, 0x12bfdc0, 0xc4200ae6c0, 0xc4204672c0)
        /home/cyphar/.local/src/github.com/kubernetes-incubator/cri-o/vendor/google.golang.org/grpc/server.go:419 +0xab
created by github.com/kubernetes-incubator/cri-o/vendor/google.golang.org/grpc.(*Server).serveStreams.func1
        /home/cyphar/.local/src/github.com/kubernetes-incubator/cri-o/vendor/google.golang.org/grpc/server.go:420 +0xa3
```

**Comment by runcom** [comment]
This seems like we're returning `nil` somewhere in the grpc responses. The last nil was from execresponse but maybe there's some code path where we return nil as well

**Comment by mrunalp** [comment]
Is this with master?

**Comment by cyphar** [comment]
Yes, I hit this with trying to run #445 on my machine.

**Comment by mrunalp** [comment]
Points to this line. 
```
if !req.GetConfig().GetLinux().GetSecurityContext().GetNamespaceOptions().HostPid && !req.GetConfig().GetLinux().GetSecurityContext().GetNamespaceOptions().HostIpc
```
Could you add some debugs to see what is nil in that line?

**Comment by cyphar** [comment]
Ah okay I figured it out, it's because I was accidentally doing this:

```
% sudo ocic pod run --config test/testdata/container_config.json --name sandbox0
```

It's quite disturbing that that's enough to cause us to die...

Here's the debug output:

```
"&PodSandboxConfig{Metadata:&PodSandboxMetadata{Name:sandbox0,Uid:,Namespace:,Attempt:1,},Hostname:,LogDirectory:,DnsConfig:nil,PortMappings:[],Labels:map[string]string{batch: no,type: small,},Annotations:map[string]string{daemon: ocid,owner: dragon,},Linux:&LinuxPodSandboxConfig{CgroupParent:,SecurityContext:nil,},}"
 ->"&LinuxPodSandboxConfig{CgroupParent:,SecurityContext:nil,}"
  ->"nil"
   ->"nil"
```

**Comment by runcom** [comment]
I remember discussing nilness checks in the requests and we agreed it was too cumbersome and just trust the kubelet or otherwise fail hard. Would we revisit this decision? :)

**Comment by mrunalp** [comment]
We can add the nil checks as well as fail if we don't get expected settings.

**Comment by runcom** [comment]
@cyphar could you test this our against master?

**Comment by rhatdan** [comment]
@cyphar I believe this must be fixed, can you close the issue?

**Comment by wking** [comment]
Fixed by the just-merged #1255?
